R/areaplot.R

```diff
@@ -141,9 +142,17 @@ AreaPlot <- function(
         datas <- split(data, data[[split_by]])
         # keep the order of levels
         datas <- datas[levels(data[[split_by]])]
+        palette <- check_palette(palette, names(datas))
+        palcolor <- check_palcolor(palcolor, names(datas))
     } else {
         datas <- list(data)
+        palette <- list(palette)
```

Comment:
We should use `as.list()`, in case one specified named vector, say `c(... = "Paired")`. `list()` will turn it to `list(c(... = "Paired")), but `as.list()` should turn it to `list(... = "Paired")`